angular-workspace/projects/ni/nimble-angular/table-column/anchor/nimble-table-column-anchor-router-link-with-href.directive.ts

```diff
@@ -28,11 +31,54 @@ export class NimbleTableColumnAnchorRouterLinkWithHrefDirective extends RouterLi
             return;
         }
 
-        // Let the router handle this navigation
-        super.routerLink = href;
-        if (!this.onClick(clickEvent.button, clickEvent.ctrlKey, clickEvent.shiftKey, clickEvent.altKey, clickEvent.metaKey)) {
+        if (!this.handleLinkClick(href, clickEvent.button, clickEvent.ctrlKey, clickEvent.shiftKey, clickEvent.altKey, clickEvent.metaKey)) {
             clickEvent.preventDefault();
         }
-        super.routerLink = null;
+    }
+
+    private handleLinkClick(url: string, button: number, ctrlKey: boolean, shiftKey: boolean, altKey: boolean, metaKey: boolean): boolean {
+        // eslint-disable-next-line @typescript-eslint/dot-notation
+        const router = this['router'] as Router | null;
+        // eslint-disable-next-line @typescript-eslint/dot-notation
+        const locationStrategy = this['locationStrategy'] as LocationStrategy;
```

Comment:
Why can't these be injected in like in the base class? Accessing private properties like this should be an extreme last resort. As you may expect, extreme last resorts probably need comments or explanation.

Reply:
I can get them via injection instead. And, actually, it seems like we can never get a null Router instance, so we don't need to null check it (and don't need the reference).